src/system.hpp

```diff
@@ -117,6 +117,18 @@ int disableFloatingPointExceptions( int const exceptions = getDefaultFloatingPoi
  */
 void setFPE();
 
+/**
+ * @brief Retieves current available memory on host
+ * @return the available memory in bytes.
+ */
+size_t getAvailableMemoryOnHost();
```

Comment:
I think I'd prefer `getAvailableMemory(MemorySpace const space)` instead of two separate functions.